File: app/src/main/java/me/sunhapper/spcharedittool/emoji/DefaultGifEmoji.java
package me.sunhapper.spcharedittool.emoji;

import android.content.Context;
import android.graphics.drawable.Drawable;
import java.io.File;
import java.io.IOException;
import me.sunhapper.spcharedittool.R;
import me.sunhapper.spcharedittool.drawable.RefreshGifDrawable;

public class DefaultGifEmoji implements Emoji {

  private String emojiText;
  private File emojiconFile;

  public DefaultGifEmoji(File emojiconFile, String emojiText) {
    this.emojiconFile = emojiconFile;
    this.emojiText = emojiText;
  }


  @Override
  public Drawable getDrawable(Context context) {
    try {
      return new RefreshGifDrawable(emojiconFile);
    } catch (IOException e) {
      e.printStackTrace();
    }
    return context.getResources().getDrawable(getDefaultResId());
  }

  public CharSequence getEmojiText() {
    return emojiText;
  }


  @Override
  public Object getRes() {
    return emojiconFile;
  }

  @Override
  public int getResType() {
    return 0;
  }

  @Override
  public boolean isDeleteIcon() {
    return false;
  }

  @Override
  public int getDefaultResId() {
    return R.drawable.common_emoj_smile_default;
  }

  @Override
  public Object getCacheKey() {
    if (emojiconFile.exists()) {
      return emojiconFile.getAbsolutePath();
    } else {
      return R.drawable.common_emoj_smile_default;
    }
  }


}

File: app/src/main/java/me/sunhapper/spcharedittool/emoji/DefaultGifEmoji.kt
package me.sunhapper.spcharedittool.emoji

import android.content.Context
import android.graphics.drawable.Drawable
import android.support.v4.content.ContextCompat
import com.sunhapper.x.spedit.gif.drawable.ProxyDrawable
import me.sunhapper.spcharedittool.R
import pl.droidsonroids.gif.GifDrawable
import java.io.File
import java.io.IOException

class DefaultGifEmoji(private val emojiconFile: File, override val emojiText: String) : Emoji {


    override val res: Any
        get() = emojiconFile


    override val isDeleteIcon: Boolean
        get() = false

    override val defaultResId: Int
        get() = R.drawable.common_emoj_smile_default

    override val cacheKey: Any
        get() = if (emojiconFile.exists()) {
            emojiconFile.absolutePath
        } else {
            R.drawable.common_emoj_smile_default
        }


    override fun getDrawable(context: Context): Drawable {
        try {
            return ProxyDrawable(GifDrawable(emojiconFile))
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return ContextCompat.getDrawable(context, defaultResId)!!
    }


}
